libs/asset-viewer/src/lib/components/map/map.component.ts

```diff
@@ -1,514 +1,205 @@
-    map((s) => s.isMapInitialised),
-    distinctUntilChanged(),
-    shareReplay({ bufferSize: 1, refCount: true })
-  );
+  readonly assetsHover = new EventEmitter<number[]>();
 
+  /**
+   * Event emitted once the map is fully initialized.
+   */
   @Output()
-  public assetClicked = new EventEmitter<number[]>();
+  readonly initializeEnd = new EventEmitter<void>();
 
-  @Input()
-  public set searchPolygon$(value: Observable<O.Option<LV95[]>>) {
-    this.state.connect('polygon', value);
-  }
+  @ViewChild('map', { static: true })
+  mapElement!: ElementRef<HTMLDivElement>;
 
-  @Input() set highlightAssetStudies(value: Observable<O.Option<number>>) {
-    this.state.connect('highlightAssetStudies', value);
-  }
+  @ViewChild('mapControls', { static: true })
+  controlsElement!: ElementRef<HTMLElement>;
 
-  private readonly studiesFromSearch$ = this._store.select(selectStudies);
-  private readonly currentAssetDetail$ = this._store.select(selectCurrentAssetDetail);
+  private readonly store = inject(Store<AppState>);
 
-  constructor() {
-    this.state.set(initialMapState);
+  private controller!: MapController;
 
-    this.state.connect('studiesFromSearch', this.studiesFromSearch$);
-    this.state.connect('currentAssetDetail', this.currentAssetDetail$);
+  controls!: {
+    zoom: ZoomControl;
+    draw: DrawControl;
+  };
 
-    this._lc.afterViewInit$
-      .pipe(take(1), subscribeOn(asyncScheduler), untilDestroyed(this))
-      .subscribe(() => this._ngAfterViewInit());
-  }
+  isInitialized = false;
 
-  _ngAfterViewInit(): void {
-    const raster = new TileLayer({
-      source: new XYZ({
-        url: `https://wmts.geo.admin.ch/1.0.0/ch.swisstopo.pixelkarte-farbe/default/current/3857/{z}/{x}/{y}.jpeg`,
-      }),
-    });
+  private readonly allStudyService = inject(AllStudyService);
 
-    const origin = {
-      center: [900000, 5900000],
-      zoom: 9,
-    };
-    const view = new View({
-      projection: 'EPSG:3857',
-      ...origin,
-      minZoom: 7,
-    });
+  private readonly subscription = new Subscription();
 
-    setTimeout(() => {
-      fitToSwitzerland(view, false);
-      const zoom = view.getZoom();
-      if (zoom != null) {
-        view.setMinZoom(zoom);
-      }
+  constructor() {
+    this.initializeEnd.subscribe(() => {
+      this.isInitialized = true;
     });
+  }
 
-    const vectorSourceAllStudies = new VectorSource({ wrapX: false });
-    const vectorLayerAllStudies = new VectorLayer({ source: vectorSourceAllStudies, minZoom: 11 });
-
-    const heatmapClusterSource = new VectorSource({ wrapX: false });
-    const source = new Cluster({
-      distance: 2,
-      source: heatmapClusterSource,
-    }) as unknown as VectorSource<Point>;
-    const heatmapLayer = new Heatmap({
-      source,
-      weight: (feature) => feature.get('features').length,
-      maxZoom: 12,
-      blur: 20,
-      radius: 5,
-      opacity: 0.7,
+  ngAfterViewInit(): void {
+    asapScheduler.schedule(() => {
```

Comment:
Was ist der Unterschied zu `setTimeout`?

Reply:
Grundsätzlich sind sie austauschbar, aber `setTimeout` kann je nach Situation & Browser einen kleinen Delay haben (siehe die [RxJS Docs](https://rxjs.dev/api/index/const/asapScheduler)). Ausserderm führt `asapScheduler` die Funktionen in der gleichen Reihenfolge aus, in der sie erfasst wurden, was `setTimeout` nicht garantiert.